src/custom/state/claim/hooks/utils.ts

```diff
@@ -151,3 +154,25 @@ function _repoNetworkIdMapping(id: SupportedChainId): string {
 export function getClaimKey(account: string, chainId: number): string {
   return `${chainId}:${account}`
 }
+
+export type PaidClaimTypeToPriceMap = {
+  [type in ClaimType]: Price<Currency, Currency> | undefined
+}
+
+/**
+ * Helper function to get vCow price based on claim type and chainId
+ *
+ * @param type
+ */
+export function mapTypeToPrice(type: ClaimType, chainId: SupportedChainId) {
+  const map: PaidClaimTypeToPriceMap = {
+    [ClaimType.Advisor]: undefined, // free
+    [ClaimType.Airdrop]: undefined, // free
+    [ClaimType.Team]: undefined, // free
+    [ClaimType.GnoOption]: GNO_PRICE,
+    [ClaimType.Investor]: USDC_PRICE,
+    [ClaimType.UserOption]: NATIVE_TOKEN_PRICE[chainId],
+  }
+
+  return map[type]
+}
```

Comment:
Why do you need a function for that?
Can't you just expose the map?
It's all static AFAICT

Reply:
chainId no?

Reply:
hmmm ok

Reply:
It still feels wrong for some reason.
You are building a map to pick only one item of that map.

What about a `switch`?

Reply:
i agree, i built off of what was there and i dont like it idk

Reply:
Coolio, let's refactor in another PR, this one already does what it needs to

Reply:
done